### ISSUE TITLE
Implement an "easy mode" class

### ISSUE THREAD

**Issue by leotrs** [ComplexGroupInteractions/xgi#37]
The proposal is to have a subclass of Hypergraph where every time the user does something fishy, it raises a warning.
```python
H = some_hypergraph()
H.add_edge([1, 1, 2])      # an edge with a repeated node is added

easy = xgi.HypergraphWithWarnings(H)
easy.add_edge([1, 1, 2])
# -> Warning: An edge contains a repeated node
```
Implementing such a class would require executing many sanity checks each time an edge is added. This makes the class slow and inefficient. However, it is very safe and it "protects" the user at each step of the way. There would also be a way to easily convert one class to the other. The main caveat would be to make sure the user remembers to convert from one class to the other when they need to. We could perhaps highlight this in the documentation, and also include it in the warning text:
```python
easy.add_edge([1, 1, 2])
# -> Warning: An edge contains a repeated node (call XXX method to disable these warnings)
```

**Comment by leotrs** [comment]
I think this is best left until after #36 is implemented.

**Comment by leotrs** [comment]
I still like this idea but will note that it has (very) low priority.

**Comment by leotrs** [comment]
With everything that's going on, I think it's safe to say we can close this issue and reopen it if the need for an easy-mode class ever arises again.
